lib/vkloader.c

```diff
@@ -146,38 +169,179 @@ ktxVulkanDeviceInfo_Construct(ktxVulkanDeviceInfo* This,
                               VkQueue queue, VkCommandPool cmdPool,
                               const VkAllocationCallbacks* pAllocator)
 {
-    VkCommandBufferAllocateInfo cmdBufInfo = {
-        .sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO
-    };
-    VkResult result;
+   return ktxVulkanDeviceInfo_ConstructEx(This, VK_NULL_HANDLE, physicalDevice, device, queue,
+                                          cmdPool, pAllocator, NULL);
+}
 
-#if defined(KTX_USE_FUNCPTRS_FOR_VULKAN)
-    // Delay loading not supported so must do it ourselves.
-    if (!ktxVulkanModuleHandle) {
-        ktx_error_code_e kresult = ktxLoadVulkanLibrary();
-        if (kresult != KTX_SUCCESS)
-            return kresult;
-    }
-#endif
+/**
+ * @memberof ktxVulkanDeviceInfo
+ * @~English
+ * @brief Construct a ktxVulkanDeviceInfo object.
+ *
+ * Records the device information, allocates a command buffer that will be
+ * used to transfer image data to the Vulkan device and retrieves the physical
+ * device memory properties for ease of use when allocating device memory for
+ * the images.
+ *
+ * Pass a valid ktxVulkanDeviceInfo* to any Vulkan KTX image loading
+ * function to provide it with the information.
+ *
+ * @param  This            pointer to the ktxVulkanDeviceInfo object to
+ *                        initialize.
+ * @param  physicalDevice handle of the Vulkan physical device.
+ * @param  device         handle of the Vulkan logical device.
+ * @param  queue          handle of the Vulkan queue.
+ * @param  cmdPool        handle of the Vulkan command pool.
+ * @param  pAllocator     pointer to the allocator to use for the image
+ *                        memory. If NULL, the default allocator will be used.
+ * @param pFunctions      pointer to the struct of functions used for
```

Comment:
Missing end of the description plus align `pFunctions` with above parameter names.